### PR TITLE
FOAM build enhancements

### PR DESCRIPTION
**Pull request by lchanmann**
## Changes
- Update pom.xml dependencies (optional)
- Support multiple deployment journals build
- Support java debugging
- Fix NPE when running -c build due to journal.0 files was generated after buildJava task

### CODE REVIEW

File: src/foam/java/refinements.js [hunk 1]
```diff
@@ -2288,9 +2288,8 @@ foam.CLASS({
         // this.fs_.writeFileSync("/tmp/" + of + ".new", this.fs_.readFileSync(outfile).toString());
 
         this.fs_.writeFileSync(outfile, javaSource);
-
-        X.javaFiles.push(outfile);
       }
+      X.javaFiles.push(outfile);
```
Comment (kgrgreer) [inline src/foam/java/refinements.js]: Why did you move this line here? The logic is so that javaFiles only contains a list of files which have been updated, not all files.

File: tools/genjava.js [hunk 2]
```diff
@@ -210,9 +210,12 @@ function processPOM(pom) {
...
 
-//console.log(X.javaFiles);
-fs_.writeFileSync('./target/javaFiles', X.javaFiles.join('\n') + '\n');
-fs_.writeFileSync('journalFiles',       X.journalFiles.join('\n') + '\n');
+// Only overwrite javaFiles when genjava:true
+if ( flags.genjava )
+  fs_.writeFileSync('./target/javaFiles', X.javaFiles.join('\n') + '\n');
+
+// REVIEW: outputJournals() should already generate all journal.0 files, writing to journalFiles is not needed
+// fs_.writeFileSync('journalFiles',       X.journalFiles.join('\n') + '\n');
```
Comment (kgrgreer) [inline tools/genjava.js]: I don't see that outputJournals() outputs these, but this was just for backward compatibility and isn't actually needed.

File: src/foam/java/refinements.js (hunk 1 above)
Comment (lchanmann) [reply src/foam/java/refinements.js]: I wanted to recompile because I was changing the generated java code but haven't found a way to only pick the files that were changed. I'll revert this as it makes the build slower.

File: tools/genjava.js (hunk 2 above)
Comment (lchanmann) [reply tools/genjava.js]: The last line of addJournal() writes the filename before its content to journal.0 file. Couldn't find the place that use the `javaFiles` file, maybe safe to stop writing out the file but I can revert if we want to keep the behavior, @kgrgreer ?
